week1/homework/src/server.js

```diff
@@ -9,7 +9,34 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    response.setHeader('Content-Type', 'application/json', 'charset = utf - 8');
+    switch (request.url) {
+      case '/state':
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/add':
+        state += 1;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/subtract':
+        state -= 1;
+        response.write(JSON.stringify({ state }));
+        break;
+      case '/reset':
+        state = 10;
+        response.write(JSON.stringify({ state }));
+        break;
+      default:
+        if (request.url === '/') {
+          response.write(JSON.stringify({ state }));
+        }
+        else {
+          response.statusCode = 404;
+          const notFound = { 'error': 'Not found' };
+          response.end(JSON.stringify(notFound, null, 2));
```

Comment:
You don't really need the `null, 2`, arguments: you could use `JSON.stringify(notFound)` the other arguments are only for formatting the output, but it makes no difference to the browser who receives the JSON response.

Reply:
Hi @J05J4 Thanks a lot for your feedback I will follow it for the next homeworks.
 see you!!